### ISSUE TITLE
[Athens, Greece] Add GTFS data for Athens public transport (https://data.hcap.gr/)

### ISSUE THREAD

**Issue by saligari** [public-transport/transitous#470]
I've come across this site that seems to be up to date.

Here: https://catalog.hcapdata.gr/dataset/dromologia-osy are the buses & trolley buses information and here https://catalog.hcapdata.gr/dataset/dromologia-statheron-sygkinonion the metro/electric train as well as tram info.

I'm not familiar with GTFS or how this stuff works or I'd file a pull request. If someone is more knowledgeable or has the time, please give this a try.

Thanks!

**Comment by saligari** [comment]
I sent them a message using their "ask for data" form as I didn't see any other contact info. Will reply again when/if they respond.

**Comment by CommanderStorm** [comment]
@saligari did they respond to your request?

I looked into if cleaning up these is possible (for a student I am working with having greece data would be interesting).

For the first dataset it is, for the second it is not.

The reason is that the second dataset is barely exists..
![Image](https://github.com/user-attachments/assets/07a6e0fa-82ed-4cf0-9043-357b2a3291c4)

Also for said dataset `calendar.txt` does not exist or failed to upload on their end.

I don't think that just the first dataset would be really usefull (or would it?).
Here is the code to clean up both and remove the random junk (this is a 2-in-1 dataset with .rar, .zip and random characters thrown in for fun)

<details><summary>hacky bash code to extract their data (click to expand)</summary>
<p>

```bash
#!/bin/bash

set -euo pipefail

if [ "$#" -ne 2 ]; then
  echo "Usage: $0 <dataset_url> <output_file>"
  echo "Example: $0 https://catalog.hcapdata.gr/dataset/dromologia-osy.jsonld dromologia-osy.zip"
  echo "Example: $0 https://catalog.hcapdata.gr/dataset/dromologia-statheron-sygkinonion.jsonld dromologia-statheron-sygkinonion.zip"
  exit 1
fi
dataset_url=$1
output_file=$2
previous_dir=`pwd`

dir=`mktemp --directory`
echo "--------------------"
echo "Downloading to $dir"
echo "--------------------"
cd $dir
curl $dataset_url \
  | jq -r '.["@graph"][] | select(.["@type"] == "dcat:Distribution") | ."dcat:accessURL"."@id"'\
  | while read -r url; do wget -q --show-progress $url; done

echo "--------------------"
echo "Cleaning up packaging mistakes $dir"
echo "--------------------"

# cleanup of the random files
unp *.rar && rm *.rar
# for some reason, they all have the underscore as first char...
sed -i 's/^_//' *.txt
mkdir gtfs1
mv *.txt gtfs1

max_calendar1=$(awk -F, 'NR > 1 {gsub(/"/, "", $2); if ($2 > max) max=$2} END {print max}' gtfs1/calendar_dates.txt)
if [[ -f "gtfs1/calendar.txt" ]]; then
    echo "using calendar.txt as it exists"
    max_calendar1=$(awk -F, 'NR > 1 { if ($NF > max) max = $NF } END { print max }' gtfs1/calendar.txt)
fi
# cleanup of ./-.zip
unzip ./-.zip -d gtfs2 && rm ./-.zip
# for some reason, their zip contains zips
find gtfs2 -type f -exec mv -t gtfs2 {} +
find . -empty -type d -delete
find gtfs2 -type f -name "*.zip" -exec unzip {} -d gtfs2 \;
find . -type f -name "*.zip" -delete
max_calendar2=$(awk -F, 'NR > 1 {gsub(/"/, "", $2); if ($2 > max) max=$2} END {print max}' gtfs2/calendar_dates.txt)
if [[ -f "gtfs2/calendar.txt" ]]; then
    echo "using calendar.txt as it exists"
    max_calendar2=$(awk -F, 'NR > 1 { if ($NF > max) max = $NF } END { print max }' gtfs2/calendar.txt)
fi

max_calendar1=$(echo "$max_calendar1" | tr --delete --complement [:digit:])
max_calendar2=$(echo "$max_calendar2" | tr --delete --complement [:digit:])
echo "max_calendar2: >$max_calendar2<"
echo "max_calendar1: >$max_calendar1<"
# the calendar of them is pretty different, but we don't want to assume which one is newer
if [[ $max_calendar1 -gt $max_calendar2 ]]; then
  echo "choosing gtfs1"
  cd gtfs1
  zip -9 ../gtfs.zip *
  cd ..
else
  echo "choosing gtfs2"
  cd gtfs2
  zip -9 ../gtfs.zip *
  cd ..
fi
rm -fr gtfs1 gtfs2

cd $previous_dir
mv $dir/gtfs.zip $output_file
echo "--------------------"
echo "Done: $output_file"
echo "--------------------"
rm -fr $dir
```

</p>
</details>

**Comment by saligari** [comment]
Nope, I e-mailed them, used the contact form & asked them about it on the telephone (they told me to e-mail them...). No response.

I do see that they've uploaded a .zip file - which for some reason contains some .txt files as .zips.

For the second link it is natural that data would barely exist, it's the metro/train lines and they don't upload time intervals anywhere. It's useful only as routes/stops and to check starting & end times, perhaps good for a programme to calculate a route with public transport. Trains pass every ~5-8mins on weekdays and every ~10-15 Saturday-Sunday.

**Comment by martin-mito** [comment]
Routing in Athens doesn't work anymore. Is the data broken?

**Comment by jbruechert** [comment]
Yes, the data is consistently published too late and doesn't include a feed_end_date, so the current implementation of looping the old data does not work.
It depends on being able to know whether it's just that specific trip that stops operating on that date or the full feed.
That can of course be improved, but the Athens data is still questionable: it says there is no public transport on Mondays and Tuesdays.

**Comment by MartinLangbecker** [comment]
I noticed that the current data is sourced from catalog.hcapdata.gr rather than the official government portal [data.gov.gr](https://data.gov.gr). The latter has an updated OSY GTFS (valid 2026-05-25 to 2026-08-25, uploaded June 3): https://data.gov.gr/dataset/dromologia-odikon-sygkoinonion-leoforeia-trolei

The HCAP/Growthfund catalog (catalog.hcapdata.gr / catalog.growthfund.gr) referenced in this issue seems to be unreachable right now. It returns 504 Gateway Timeout. [data.gov.gr](https://data.gov.gr) provides the same data at a stable URL that seemingly doesn't change between uploads, so maybe that could be an easy fix.

**Comment by jbruechert** [comment]
Sounds good, looks like the situation finally improved! We've been debugging this particular feed quite a number of times by now.

**Comment by Lach-anonym** [comment]
Good it moves somewhere there!

**Comment by MartinLangbecker** [comment]
I opened PR #2208 to update the data source.

Regarding the rail feed (STASY, metro/ISAP/tram): there's also a GTFS on data.gov.gr (dromologia-statheron-sygkoinonion-metro-isap-tram) with a stable URL, currently valid through June 30. However, it uses only calendar_dates.txt (no calendar.txt), so extend-calendar won't help when it expires. Given that updates tend to come late, this will likely have recurring coverage gaps. I left it out of #2208 for now.
